File: 1920.build-array-from-permutation.java
/*
 * @lc app=leetcode id=1920 lang=java
 *
 * [1920] Build Array from Permutation
 */

// @lc code=start
class Solution {
    public int[] buildArray(int[] nums) {
        final int mask = 0x03ff;
        for (int ix = 0; ix != nums.length; ++ix) {
            nums[ix] = nums[ix] | ((nums[nums[ix]] & mask) << 10);
        }
        for (int ix = 0; ix != nums.length; ++ix) {
            nums[ix] = (nums[ix] & (~mask)) >> 10;
        }
        return nums;
    }
}
// @lc code=end

File: 1920.build-array-from-permutation.cpp
/*
 * @lc app=leetcode id=1920 lang=cpp
 *
 * [1920] Build Array from Permutation
 */

// @lc code=start
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
        vector<int> ret(nums.size());
        for (auto ix = 0ull; ix != nums.size(); ++ix) ret[ix] = nums[nums[ix]];
        return ret;
    }
};
// @lc code=end
